roles/node_exporter/templates/node_exporter.service.j2

```diff
@@ -8,6 +8,16 @@ After=network-online.target
 Type=simple
 User={{ node_exporter_system_user }}
 Group={{ node_exporter_system_group }}
+{% if node_exporter_textfile_dir | length > 0 %}
+{% if (ansible_facts.packages.systemd | first).version is version('235', '>=') %}
+ExecStartPre=+/bin/mkdir -p {{ node_exporter_textfile_dir }}
+ExecStartPre=+/bin/chown -R {{ node_exporter_system_user }}:{{ node_exporter_system_group }} {{ node_exporter_textfile_dir }}
```

Comment:
Hmm, I wonder if this should be `-R`. I'm not sure we want all of the prom files changed, just the dir.

Reply:
In our environment, the text file exporters are started after `node_exporter.service` and the directory is used exclusively for text file exporters, so it doesn't really matter, but I think we can remove the `-R`, or should we go further and only change the owner if the directory is empty?